chemprop/data/data.py

```diff
@@ -179,7 +179,11 @@ def __init__(self,
     @property
     def mol(self) -> List[Union[Chem.Mol, Tuple[Chem.Mol, Chem.Mol]]]:
         """Gets the corresponding list of RDKit molecules for the corresponding SMILES list."""
-        mol = make_mols(self.smiles, self.is_reaction_list, self.is_explicit_h_list, self.is_adding_hs_list, self.is_keeping_atom_map_list)
+        if self.atom_targets is not None or self.bond_targets is not None:
+            # When the original atom mapping is used, the explicit hydrogens specified in the input SMILES are used
+            mol = make_mols(self.smiles, self.is_reaction_list, self.is_keeping_atom_map_list, self.is_adding_hs_list, self.is_keeping_atom_map_list)
+        else:
+            mol = make_mols(self.smiles, self.is_reaction_list, self.is_explicit_h_list, self.is_adding_hs_list, self.is_keeping_atom_map_list)
```

Comment:
see previous comment. Here you are passing differently, and I think we need some comments to explain the difference. Otherwise the previous implementation looks like a bug.

Reply:
In version 1.6.0, explicit Hs will be considered when `--keeping_atom_map` is used. However, there was a bug related to scaffold splits at that time, so the behavior has changed un v1.6.1. Since we didn't have a test for using `--keeping_atom_map` at that time, the previous PR broke this function. Thus, we need to find some way to force the explicit Hs to be added back when `--keeping_atom_map` is used.